src/metrics/representation.py

```diff
@@ -0,0 +1,381 @@
+            db_ids: Database indexes.
+            search_batch_size: Query size for one search request.
+            k:  Number of top closest indexes to get.
+
+        Returns:
+            Generator wich contain relevant and closest Tuple values. 
+
+            Relevant include relevant indexes and scores, size (search_batch_size, , 2).
+            Closest include searched indexes and distances, size (search_batch_size, , 2).
+        """
+        def generator():
+            """Generate relevant - y_true, and closest - y_pred for metric calculation with ranx library.
+
+            Returns:
+                relevant: List of relevant indexes with its scores per each queries. Length of list = search_batch_size.
+                    And size of one element of list = (relevant_indexes_size, 2), where last shape 2 for relevant index
+                    and it score. 
+                    Example for 3 search_batch_size, and relevant_sizes = [2, 2, 1] with score = 1 for every \
+                    relevant index:
+                    [
+                        np.array([[6, 1], [7, 1]]), 
+                        np.array([[2, 1], [5, 1]]), 
+                        np.array([[4, 1]])
+                    ].
+                closest: List of numpy arrays, with nearest searched indexes by top k, with its distances.
+                    Example for k = 3:
+                    [
+                        np.array([[4.        , 0.29289323],
+                               [2.        , 0.29289323],
+                               [6.        , 0.42264974]]), 
+                        np.array([[5.        , 0.29289323],
+                               [2.        , 0.29289323],
+                               [6.        , 0.42264974]]), 
+                        np.array([[4.        , 0.29289323],
+                               [5.        , 0.29289323],
+                               [6.        , 0.42264974]])
+                    ].
+            """
+            for i in range(0, len(queries), search_batch_size):
+                if i + search_batch_size >= len(queries):
+                    query_idxs = np.arange(i, len(queries))
+                else:
+                    query_idxs = np.arange(i, i + search_batch_size)
+                    
+
+                closest_dist, closest_idx = index.search(queries[query_idxs], k=k)
+                relevant = relevants[query_idxs]
+ 
+                if self.metric == 0:
+                    closest_dist = 1 - closest_dist
+            
+                closest = map(lambda idx: np.stack((db_ids[closest_idx[idx]], closest_dist[idx]), axis=1), \
+                    np.arange(len(closest_idx)))
+                
+                if scores is None:
+                    relevant = map(lambda r: np.stack((r, np.ones_like(r)), axis=1), relevant)
+                else:
+                    relevant = map(lambda r_q: np.stack((r_q[0], scores[r_q[0], q_order_idxs[r_q[1]]]), axis=1), \
```

Comment:
Shouldn't scores be filtered like this? 
```
query_scores = scores[r_q[0], q_order_idxs[r_q[1]]]
rel_scores = query_scores[query_scores > 0.]
```

Reply:
It's already done in function `self.prepare_representation_data`.